virtool/samples/data.py

```diff
@@ -30,18 +31,18 @@
 )
 from virtool.samples.oas import CreateSampleSchema, EditSampleSchema
 from virtool.samples.utils import SampleRight, join_sample_path
-from virtool.settings.db import get_settings
 from virtool.subtractions.db import AttachSubtractionTransform
 from virtool.uploads.db import reserve
 from virtool.users.db import AttachUserTransform
 from virtool.utils import base_processor, run_in_thread, wait_for_checks
 
 
 class SamplesData(DataLayerPiece):
-    def __init__(self, config: Config, db, pg: AsyncEngine):
-        self._config = config
-        self._db = db
-        self._pg = pg
+    def __init__(self, app):
+        self._app = app
+        self._config: Config = app["config"]
+        self._db = app["db"]
+        self._pg: AsyncEngine = app["pg"]
```

Comment:
The settings should be available at `self.data.settings`.

Explanation:
* https://github.com/virtool/virtool/blob/main/virtool/data/piece.py
* https://github.com/virtool/virtool/blob/main/virtool/data/layer.py